File: src/test/java/leetcode/MinStepsToMakeTwoStringsAnagramTest.java
package leetcode;

import static org.assertj.core.api.Assertions.assertThat;

import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.CsvSource;

class MinStepsToMakeTwoStringsAnagramTest {

  @ParameterizedTest
  @CsvSource(
      delimiter = '|',
      textBlock =
          """
      bab | aba | 1
      leetcode | practice | 5
      anagram | mangaar | 0
      """)
  void minSteps(String s, String t, int expected) {
    MinStepsToMakeTwoStringsAnagram sut = new MinStepsToMakeTwoStringsAnagram();
    int actual = sut.minSteps(s, t);
    assertThat(actual).isEqualTo(expected);
  }
}

File: leetcode-kt/src/test/kotlin/com/giwankim/leetcode/MinStepsToMakeTwoStringsAnagramTest.kt
package com.giwankim.leetcode

import io.kotest.core.spec.style.FunSpec
import io.kotest.datatest.withTests
import io.kotest.matchers.shouldBe

class MinStepsToMakeTwoStringsAnagramTest :
    FunSpec({
        val sut = MinStepsToMakeTwoStringsAnagram()

        context("min steps to make two strings anagram") {
            withTests(
                nameFn = { (s, t, expected) -> "s=$s, t=$t, expected=$expected" },
                MinStepsToMakeTwoStringsAnagramCase("bab", "aba", 1),
                MinStepsToMakeTwoStringsAnagramCase("leetcode", "practice", 5),
                MinStepsToMakeTwoStringsAnagramCase("anagram", "mangaar", 0),
            ) { (s, t, expected) ->
                sut.minSteps(s, t) shouldBe expected
            }
        }
    })

private data class MinStepsToMakeTwoStringsAnagramCase(
    val s: String,
    val t: String,
    val expected: Int,
)
